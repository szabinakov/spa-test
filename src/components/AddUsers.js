import React, { useState } from "react";
import axios from "axios";
import Alert from "./Alert.js";
import "../styles/Form.css";

export default function AddUsers() {
  const initialState = {
    fields: {
      firstname: "",
      lastname: "",
      job: "",
      email: "",
    },
    alert: {
      message: "",
      isSuccess: false,
    },
  };
  const [fields, setFields] = useState(initialState.fields);
  const [alert, setAlert] = useState(initialState.alert);
  const handleFieldChange = (event) => {
    setFields({ ...fields, [event.target.name]: event.target.value });
  };
  const handleAddUser = (event) => {
    event.preventDefault();
    setAlert({ message: "", isSuccess: false });
    axios
      .post("https://reqres.in/api/users", fields)
      .then(() => {
        setAlert({
          message: "User has been added successfully",
          isSuccess: true,
        });
      })
      .then(setTimeout(() => setAlert({ message: "", isSuccess: false }), 2000))
      .catch(() => {
        setAlert({
          message: "Something went wrong! Please try again!",
          isSuccess: false,
        });
      });
  };
  return (
    <>
      <Alert message={alert.message} success={alert.isSuccess} />
      <div className="formContainer">
        <h2>Please fill out the details of the user you would like to add!</h2>
        <form className="form" data-testid="form" onSubmit={handleAddUser}>
          <div>
            <input
              type="text"
              className="firstname"
              data-testid="firstname"
              name="firstname"
              onChange={handleFieldChange}
              placeholder="First name!"
              required
            />
          </div>
          <div>
            <input
              type="text"
              className="lastname"
              data-testid="lastname"
              name="lastname"
              onChange={handleFieldChange}
              placeholder="Last name!"
              required
            />
          </div>
          <div>
            <input
              type="text"
              className="job"
              data-testid="job"
              name="job"
              onChange={handleFieldChange}
              placeholder="Job title!"
            />
          </div>
          <div>
            <input
              type="text"
              className="email"
              data-testid="email"
              name="email"
              onChange={handleFieldChange}
              placeholder="Email address!"
              required
            />
          </div>
          <div data-testid="buttonDiv" className="buttonDiv">
            <button type="submit">Add User</button>
          </div>
        </form>
      </div>
    </>
  );
}
