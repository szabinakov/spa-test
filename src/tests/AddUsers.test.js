import { render } from "@testing-library/react";
import AddUsers from "../components/AddUsers.js";

describe("AddUsers", () => {
  it("renders correctly", () => {
    const { asFragment } = render(<AddUsers />);

    expect(asFragment()).toMatchSnapshot();
  });

  it("it renders every element", () => {
    const { getByTestId } = render(<AddUsers />);

    expect(getByTestId("form")).toHaveClass("form");
    expect(getByTestId("firstname")).toHaveClass("firstname");
    expect(getByTestId("lastname")).toHaveClass("lastname");
    expect(getByTestId("job")).toHaveClass("job");
    expect(getByTestId("email")).toHaveClass("email");
    expect(getByTestId("buttonDiv")).toHaveClass("buttonDiv");
  });
});
