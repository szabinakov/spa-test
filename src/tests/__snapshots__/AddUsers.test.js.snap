// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddUsers renders correctly 1`] = `
<DocumentFragment>
  <div
    class="formContainer"
  >
    <h2>
      Please fill out the details of the user you would like to add!
    </h2>
    <form
      class="form"
      data-testid="form"
    >
      <div>
        <input
          class="firstname"
          data-testid="firstname"
          name="firstname"
          placeholder="First name!"
          required=""
          type="text"
        />
      </div>
      <div>
        <input
          class="lastname"
          data-testid="lastname"
          name="lastname"
          placeholder="Last name!"
          required=""
          type="text"
        />
      </div>
      <div>
        <input
          class="job"
          data-testid="job"
          name="job"
          placeholder="Job title!"
          type="text"
        />
      </div>
      <div>
        <input
          class="email"
          data-testid="email"
          name="email"
          placeholder="Email address!"
          required=""
          type="text"
        />
      </div>
      <div
        class="buttonDiv"
        data-testid="buttonDiv"
      >
        <button
          type="submit"
        >
          Add User
        </button>
      </div>
    </form>
  </div>
</DocumentFragment>
`;
